src/python/pants/backend/python/typecheck/pyright/rules_integration_test.py

```diff
@@ -287,6 +288,52 @@ def test_skip(rule_runner: PythonRuleRunner) -> None:
     assert not result
 
 
+def test_passing_cache_clear(rule_runner: PythonRuleRunner) -> None:
+    # Ensure that the requirements venv must be created, by adding in a third-party
+    # requirement to the test code.
+    rule_runner.write_files(
+        {
+            "BUILD": "python_requirement(name='more-itertools', requirements=['more-itertools==8.4.0'])",
+            f"{PACKAGE}/f.py": dedent(
+                """\
+            from more_itertools import is_sorted
+
+            assert is_sorted([1, 2, 3]) is True
+            """
+            ),
+            f"{PACKAGE}/BUILD": "python_sources()",
+        }
+    )
+    tgt = rule_runner.get_target(Address(PACKAGE, relative_file_path="f.py"))
+
+    # On the first run, it should work as advertised with no modifications.
+    result = run_pyright(rule_runner, [tgt])
+    assert len(result) == 1
+    assert result[0].exit_code == 0
+    assert "0 errors" in result[0].stdout
+    assert result[0].report == EMPTY_DIGEST
+
+    original_cache_dir = (
+        rule_runner.options_bootstrapper.bootstrap_options.for_global_scope().named_caches_dir
+    )
+
+    # Should now fail with a different cache directory, since there is no venv
```

Comment:
I don't understand this. Isn't the whole point of this change that we are not relying on the venv existing, and we will force-create it if needed?

Reply:
The right thing to simulate is deleting the named_cache, and then rerunning *without* changing any process inputs.

Reply:
And you want to manually verify that this fails if you comment out the change in rules.py, but passes if you leave it in.

Reply:
Understood on the last part - I have a separate branch that contains this test, but without the change in rules.py so I'm able to test between the two. Will rework the rest to adhere to what you mentioned.